integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java

```diff
@@ -0,0 +1,131 @@
+import java.io.IOException;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.junit.Ignore;
+import org.junit.Test;
+import static com.amazon.opendistroforelasticsearch.sql.esintgtest.TestsConstants.TEST_INDEX_ACCOUNT;
+import static org.elasticsearch.rest.RestStatus.OK;
+
+public class QueryAnalysisIT extends PPLIntegTestCase {
+
+    @Override
+    public void init() throws IOException {
+        loadIndex(Index.ACCOUNT);
+    }
+
+    /**
+     * Valid commands should pass semantic check
+     */
+    @Test
+    public void searchCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s age=20", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void whereCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | where age=20", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void fieldsCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | fields firstname", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Ignore("Can't resolve target field yet")
+    @Test
+    public void renameCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | rename firstname as first", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void statsCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | stats avg(age)", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void dedupCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | dedup firstname, lastname", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void sortCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | sort age", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void evalCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | eval age=abs(age)", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+
+    /**
+     * Commands that fail semantic analysis should throw {@link SemanticCheckException}
+     */
+    @Test
+    public void unsupportedAggregationShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | stats count(age)", TEST_INDEX_ACCOUNT);
+        queryShouldThrowSemanticException(query, "Unsupported aggregation function count");
+    }
+
+    @Test
+    public void nonexistentIndexShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | fields name", TEST_INDEX_ACCOUNT);
```

Comment:
The test method name should be nonexistent field? But I think we do need another one for index.

Reply:
It should be nonexistent field, corrected it now. It seems we have not set semantic check for index, let's add the index check when the index check is enabled.

Reply:
The `ElasticsearchIndex.getFieldTypes()` should throw Elasticsearch IndexNotFoundException. Not sure if the exception is caught without re-throw or for other reason.

Reply:
Yes Elasticsearch throws this exception. I didn't add it here because it's not handled by the semantic check. Maybe we can add this to other test

Reply:
Sounds good!